### ISSUE TITLE
About the arguments of model

### ISSUE THREAD

**Issue by meiyuan666** [anonymousgit2020/fedsem#1]
Hi , I am glad to read your paper, Multi-Center Federated Learning. The work in the paper is innovative and interesting,from which I get a lot of inspiration. However, when I reproduce your experiment, for example: run python fedcluster.py, I do not know how to choose the arguments of  "model".  Could you give me some more guidance? Thank you very much!

**Comment by sky1106** [comment]
Hi, I meet the same problem with you, if you have found the solution? Could you please give me an example of all arguments to run the fedcluster.py? Thank you every much

**Comment by sky1106** [comment]
> Hi , I am glad to read your paper, Multi-Center Federated Learning. The work in the paper is innovative and interesting,from which I get a lot of inspiration. However, when I reproduce your experiment, for example: run python fedcluster.py, I do not know how to choose the arguments of "model". Could you give me some more guidance? Thank you very much!
Hi, I meet the same problem with you, if you have found the solution? Could you please give me an example of all arguments to run the fedcluster.py? Thank you every much

**Comment by sky1106** [comment]
> Hi , I am glad to read your paper, Multi-Center Federated Learning. The work in the paper is innovative and interesting,from which I get a lot of inspiration. However, when I reproduce your experiment, for example: run python fedcluster.py, I do not know how to choose the arguments of "model". Could you give me some more guidance? Thank you very much!

Hi, I meet the same problem with you, if you have found the solution? Could you please give me an example of all arguments to run the fedcluster.py? Thank you every much

**Comment by meiyuan666** [comment]
> Hi , I am glad to read your paper, Multi-Center Federated Learning. The work in the paper is innovative and interesting,from which I get a lot of inspiration. However, when I reproduce your experiment, for example: run python fedcluster.py, I do not know how to choose the arguments of "model". Could you give me some more guidance? Thank you very much!
> 
> Hi, I meet the same problem with you, if you have found the solution? Could you please give me an example of all arguments to run the fedcluster.py? Thank you every much

I haven't found a solution yet, sorry~

**Comment by sky1106** [comment]
> > Hi , I am glad to read your paper, Multi-Center Federated Learning. The work in the paper is innovative and interesting,from which I get a lot of inspiration. However, when I reproduce your experiment, for example: run python fedcluster.py, I do not know how to choose the arguments of "model". Could you give me some more guidance? Thank you very much!
> > Hi, I meet the same problem with you, if you have found the solution? Could you please give me an example of all arguments to run the fedcluster.py? Thank you every much
> 
> I haven't found a solution yet, sorry~

OK~ thank you all the same, we can wait for the author's reply

**Comment by moFang222** [comment]
@meiyuan666 @sky1106 
Hi! Did you guys reproduce the result of FedSM algorithm in the original paper?I can't achieve its accuracy when reproducing, I wonder if this is the cause of some certain hyperparameters, because  the paper does not specify these hyperparameters in detail.

**Comment by sky1106** [comment]
> @meiyuan666 @sky1106
> Hi! Did you guys reproduce the result of FedSM algorithm in the original paper?I can't achieve its accuracy when reproducing, I wonder if this is the cause of some certain hyperparameters, because the paper does not specify these hyperparameters in detail.

Hi! You have already reproduced the algorithm? Could you please tell me the  arguments that you ran with? We can disscuss about this. By the way, did you find some bugs when you reproduced the algorithm?

**Comment by moFang222** [comment]
@sky1106 If you mean the FedSME algorithm,I ran it with code `python3 mlhead.py -dataset femnist -model cnn --clients-per-round 20 --num-epochs 5 -lr 0.003`.I haven't run the whole experiment yet, but the accuracy is almost 0.1 when I reach 30 rounds

**Comment by sky1106** [comment]
> @sky1106 If you mean the FedSME algorithm,I ran it with code `python3 mlhead.py -dataset femnist -model cnn --clients-per-round 20 --num-epochs 5 -lr 0.003`.I haven't run the whole experiment yet, but the accuracy is almost 0.1 when I reach 30 rounds

It works not for me, there are lots of bugs to be fixed. You did not meet these bugs when you reproduced the algorithm?
![bugs](https://user-images.githubusercontent.com/44168088/84004710-8f295900-a99e-11ea-99d3-ac88f29d76b3.png)

**Comment by moFang222** [comment]
@sky1106 It looks like something went wrong with the femnist data set. Did you prepare the data set as described [here](https://github.com/anonymousgit2020/fedsem/tree/b8ad27739b34c38dd5a0c3938ce846cf7aa2914b/data/femnist) ？

**Comment by sky1106** [comment]
> @sky1106 It looks like something went wrong with the femnist data set. Did you prepare the data set as described [here](https://github.com/anonymousgit2020/fedsem/tree/b8ad27739b34c38dd5a0c3938ce846cf7aa2914b/data/femnist) ？

OK~I still meet lots of bugs, I will fix them first and then tell you my results. Thank you very much!

**Comment by moFang222** [comment]
@sky1106 ok!And please try some different hyperparameters than those I listed above to make sure you find the hyperparameters that produce the best result!Waiting for your result！

**Comment by meiyuan666** [comment]
@moFang222 @sky1106 Thank you for all your replies!  And have you already reproduced the experiment of FedCluster?

**Comment by moFang222** [comment]
> @moFang222 @sky1106 Thank you for all your replies! And have you already reproduced the experiment of FedCluster?

@meiyuan666 Sorry, I didn't see FedCluster in the paper, do you mean FedSEM algorithm?Or the Clustered baseline algorithm in the red box below.
![Snipaste_2020-06-08_19-55-05](https://user-images.githubusercontent.com/25848511/84027793-26ec6e80-a9c2-11ea-8202-fd4e762dbc9b.png)

**Comment by meiyuan666** [comment]
> > @moFang222 @sky1106 Thank you for all your replies! And have you already reproduced the experiment of FedCluster?
> 
> @meiyuan666 Sorry, I didn't see FedCluster in the paper, do you mean FedSEM algorithm?Or the Clustered baseline algorithm in the red box below.
> ![Snipaste_2020-06-08_19-55-05](https://user-images.githubusercontent.com/25848511/84027793-26ec6e80-a9c2-11ea-8202-fd4e762dbc9b.png)

Yeah, the baseline algorithm, which is proposed by the paper : Clustered federated learning: Model-agnostic distributed multi-task optimization under privacy constraints. The relevant implementation code is also in this repository:  all_baselines/fed-cluster/models/fedcluster.py.

**Comment by moFang222** [comment]
@meiyuan666 Sorry,I have not reproduced this algorithm yet.

**Comment by meiyuan666** [comment]
> @meiyuan666 Sorry,I have not reproduced this algorithm yet.

It doesn't matter.Thank you very much for your reply~

**Comment by peiji1981** [comment]
hi guys, I am also doing research about federated clustering algorithms. I wanna ask for a paper/code/idea about vertically federated clustering, did you see that ? thx you all ~
